lib/Sema/TypeCheckPattern.cpp

```diff
@@ -1286,22 +1432,20 @@ Pattern *TypeChecker::coercePatternToType(ContextualPattern pattern,
           // isn't a static VarDecl, so the existing mechanics in
           // extractEnumElement won't work.
           if (type->getAnyNominal() == Context.getOptionalDecl()) {
-            if (EEP->getName().isSimpleName("None") ||
-                EEP->getName().isSimpleName("Some")) {
+            if (EEP->getName().getBaseIdentifier().str() == "None" ||
```

Comment:
You shouldn't have to touch this code. The guard above checks if the nominal type we're dealing with is precisely `Swift.Optional<T>`

Reply:
If anything we oughta just get rid of it. These are Swift 2-3 migration guides.